### PR TITLE
Compress and Decompress

### PR DESCRIPTION
**Pull request by Nirmalperiyasamy**

### CODE REVIEW

File: archiveFile/src/main/java/org/hriday/archiveFile/ArchiveFile.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class ArchiveFile implements ArchiveMethods {
+
+    @Override
+    public void compress(String sourceFile, String compressFile) throws IOException {
+
+        String[] extension = compressFile.split("\\.");
+        String compress = extension[0] + ".zip";
```
Comment (gokul656) [inline archiveFile/src/main/java/org/hriday/archiveFile/ArchiveFile.java]: outputFileName

File: archiveFile/src/main/java/org/hriday/archiveFile/ArchiveFile.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        byte[] buffer = new byte[1024];
+
+        FileOutputStream out = new FileOutputStream(compress);
+        ZipOutputStream zipOut = new ZipOutputStream(out);
+
+        File folder = new File(sourceFile);
+        for (File multi : folder.listFiles()) {
+            if (!multi.isDirectory()) {
```
Comment (gokul656) [inline archiveFile/src/main/java/org/hriday/archiveFile/ArchiveFile.java]: rename to file

File: archiveFile/src/main/java/org/hriday/archiveFile/ArchiveFile.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                while ((len = fileInputStream.read(buffer)) > 0) {
+                    zipOut.write(buffer, 0, len);
+                }
+
+                fileInputStream.close();
+            }
+
+        }
+        zipOut.closeEntry();
+        zipOut.close();
```
Comment (gokul656) [inline archiveFile/src/main/java/org/hriday/archiveFile/ArchiveFile.java]: wrap with try catch and finally

File: convertFileSize/src/main/java/com/hriday/convertFileSize/dto/ArchiveDetailsDto.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.hriday.convertFileSize.dto;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/dto/ArchiveDetailsDto.java:10]: remove unwanted spaces

File: convertFileSize/src/main/java/com/hriday/convertFileSize/factory/ArchiveService.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+
+public interface ArchiveService {
+    String compress(MultipartFile[] file) throws IOException;
+
+//    String decompress(MultipartFile file) throws IOException;
+
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/factory/ArchiveService.java:12]: remove unwanted comments

File: convertFileSize/src/main/java/com/hriday/convertFileSize/globalException/GlobalException.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @ExceptionHandler(value = CustomException.class)
+    public ResponseEntity<?> handler(CustomException exception) {
+        return ResponseEntity.badRequest().body(exception.getMessage());
+    }
+
+    @ExceptionHandler(MaxUploadSizeExceededException.class)
+    public ResponseEntity<?> handleMaxSizeException(MaxUploadSizeExceededException exc) {
+        return ResponseEntity
+                .status(HttpStatus.EXPECTATION_FAILED)
+                .body("The file size exceeds the maximum allowed limit.");
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/globalException/GlobalException.java]: Move to constants

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    @Autowired
+    protected ArchiveRepo archiveRepo;
+
+    @Value("${fileStorage}")
+    protected String fileStoragePath;
+
+    @Value("${tempStorage}")
+    public String tempStoragePath;
+
+    ArchiveFile archiveFile = new ArchiveFile();
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: Could be changed to bean

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        }
+
+        archiveFile.compress(tempFilePath, compressedFilePath);
+
+        return logsInDatabase(compressedFile, compressedFile);
+    }
+
+    public String logsInDatabase(File tempFile, File compressedFile) {
+
+        ArchiveDetailsDto archiveDetailsDto = new ArchiveDetailsDto();
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: Could be changed to a builder class

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    ArchiveFile archiveFile = new ArchiveFile();
+
+    @Override
+    public String compress(MultipartFile[] file) throws IOException {
+
+        String fileName = StringUtils.cleanPath(Objects.requireNonNull(file[0].getOriginalFilename()));
+
+        String[] extension = fileName.split("\\.");
+
+        EnumSet<FileType> fileTypes = EnumSet.allOf(FileType.class);
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: Could be optimised using Enum.valueOf()

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+        convertMultipartFileToFile(file, tempFilePath);
+
+        String decompressedFilepath = fileStoragePath + "\\";
+
+        File decompressedFile = new File(decompressedFilepath);
+
+        archiveFile.decompress(tempFilePath, decompressedFile);
+
+        return "m";
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: check this return

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+        if (resource.exists() && resource.isReadable()) {
+            logAfterDownload(archiveDetails);
+            return resource;
+        } else {
+            throw new CustomException("the file doesn't exist or not readable");
+        }
+    }
+
+    private void logAfterDownload(ArchiveDetails archiveDetails) {
+        archiveDetails.setStatus("downloaded");
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: Can be converted to enum

Comment (Nirmalperiyasamy) [reply convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: resolved

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java (hunk 8 above)
Comment (Nirmalperiyasamy) [reply convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: changed to builder

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java (hunk 7 above)
Comment (Nirmalperiyasamy) [reply convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java]: converted to bean

File: convertFileSize/src/main/java/com/hriday/convertFileSize/controller/FileController.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+        Resource resource = fileStorageService.downloadFile(uid);
+
+        try {
+
+            ZipOutputStream zipOutputStream = new ZipOutputStream(response.getOutputStream());
+
+            ZipEntry zipEntry = new ZipEntry((Objects.requireNonNull(resource.getFilename())));
+
+            try {
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/controller/FileController.java]: Remove redundant try catch

File: convertFileSize/src/main/java/com/hriday/convertFileSize/exception/GlobalException.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+public class GlobalException {
+
+    @ExceptionHandler(value = CustomException.class)
+    public ResponseEntity<?> handler(CustomException exception) {
+        return ResponseEntity.badRequest().body(exception.getMessage());
+    }
+
+    @ExceptionHandler(MaxUploadSizeExceededException.class)
+    public ResponseEntity<?> handleMaxSizeException(MaxUploadSizeExceededException exc) {
+        return ResponseEntity
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/exception/GlobalException.java:20]: Remove `value=` order add `value=` in MaxUploadSizeExceededException.class

File: convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+        archiveRepo.save(archiveDetails);
+    }
+
+    public Resource downloadFile(String uid) throws IOException {
+
+        ArchiveDetails archiveDetails = archiveRepo.findByUid(uid);
+
+        Path path = Paths.get(fileStoragePath).toAbsolutePath().resolve(archiveDetails.getFileName() + Const.ZIP);
+        Resource resource;
+
```
Comment (gokul656) [inline convertFileSize/src/main/java/com/hriday/convertFileSize/service/FileStorageService.java:172]: wrap into same try

**Review by gokul656 — APPROVED** [review]

**Pull request merged**